### PR TITLE
Add telemetry for secret backend execution time

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Add telemetry measure for secret backend execution time.

### Motivation

We sometimes have agents failing to start because of timeouts while decoding secrets via the secret backend.
This PR allows to monitor the secret backend execution time so that we can notice when it becomes abnormally slow even if it doesn’t time out.

### Describe your test plan

Start an agent with telemetry enabled and a secret in its config:
```yaml
app_key: ENC[foo]
telemetry:
  enabled: true
```

And check the telemetry gauge:
```shell
$ docker exec -ti datadog-agent curl http://localhost:5000/telemetry | grep secret

# HELP secret_backend__elapsed_ms Elapsed time of secret backend invocation
# TYPE secret_backend__elapsed_ms gauge
secret_backend__elapsed_ms{command="/readsecret.sh",exit_code="0"} 2047
```

### CODE REVIEW

File: pkg/secrets/fetch_secret.go [hunk 1]
```diff
@@ -58,15 +65,29 @@ func execCommand(inputPayload string) ([]byte, error) {
 	cmd.Stdout = &stdout
 	cmd.Stderr = &stderr
 
+	start := time.Now()
 	err := cmd.Run()
+	elapsed := time.Since(start)
+	log.Debugf("secret_backend_command '%s' completed in %d ms", secretBackendCommand, elapsed.Milliseconds())
```
Comment (xornivore) [inline pkg/secrets/fetch_secret.go:71]: Could format time.Duration instead of Milliseconds as int here?

**Review by xornivore — APPROVED** [review]
LGTM, minor nit about time formatting in log message.

File: pkg/secrets/fetch_secret.go (hunk 1 above)
Comment (L3n41c) [reply pkg/secrets/fetch_secret.go]: Good point ! 👍🏻 
Change done: https://github.com/DataDog/datadog-agent/pull/6381/commits/982a4a1a5afc13f36fb87dbb249fab4ee158310c

**Review by remeh — COMMENTED** [review]
The new metric LGTM but I'm not sure to understand the introduction of this new package.
Did you have some cyclic import while adding this? Where was the cycle located?

**Review by remeh — APPROVED** [review]
We discussed the import cycle offline 👍

**Comment by L3n41c** [comment]
Hello @remeh ,
Indeed, I had to split the `telemetry` package and to introduce `telemetry/utils` because the functions that I moved to the `telemetry/utils` where depending on the `config` package. The `config` package depends on the `secret` package. And the change that I wanted to add in this PR was to add `telemetry` features inside the `secret` package.

**Review by mx-psi — APPROVED** [review]
Changes on Platform owned files LGTM

**Review by shang-wang — APPROVED** [review]
Looks good from processes

**Pull request merged**